### ISSUE TITLE
UnhandledPromiseRejectionWarning: TypeError: console[msg._type] is not a function

### ISSUE THREAD

**Issue by robraux** [meteortesting/meteor-browser-tests#25]
```
UnhandledPromiseRejectionWarning: TypeError: console[msg._type] is not a function
    at packages/meteortesting:browser-tests/browser/puppeteer.js:35:37
```

The issue is the msg._type is `warning`, not `warn` when dealing with react PropType validation messages. 
```page.on('console', async msg => console[msg._type](``` is not defensive.

```
let messageType = msg._type;
if(console[messageType] === undefined) {
    messageType = 'warn';
}
```

Should be enough to wrap this and prevent rejection message. I can create a PR if this is something you're interested in including.

**Comment by SimonSimCity** [comment]
Thanks for reporting this issue.

There are in fact more message types which are not handled properly (https://github.com/puppeteer/puppeteer/blob/master/docs/api.md#consolemessagetype) - which just have a different name than their counterpart on the `console` object (https://nodejs.org/api/console.html).

Others having a different name are:
* `startGroup` -> `group`
* `startGroupCollapsed` -> `groupCollapsed`
* `endGroup` -> `groupEnd`

I think the best approach would be to create a mapping for those who have a different name and add something like `[UNKNOWN TYPE ${msg.type()}]` to the beginning of the message if the name is unknown. How we should report them is something I'd leave open to you. It be argumented to be an error or it could also be just something else, because it's not an error by the website, but by the package.

A PR is most welcome, thanks for volunteering!

**Comment by SimonSimCity** [comment]
Fixed by merging #26 - released in `v1.3.2`
